packages/agent/src/agent/api.ts

```diff
@@ -133,6 +138,10 @@ export interface SubmitResponse {
     } | null;
     headers: HttpHeaderField[];
   };
+  requestDetails?: {
```

Comment:
Rather than calling it `requestDetails` and only include `ingressExpiry` and `nonce` I would just add the `contentMap` here. Is there a reason why we should _not_ reflect back `method`, `arg`, etc?
Also, it is then clear that `requestIdOf(contentMap) === requestId` must hold.

Reply:
Admittedly, I didn't know exactly what `contentMap` meant. I can return the whole `CallRequest` object so that `requestIdOf` can be recomputed